### ISSUE TITLE
Possible to work with Signal?

### ISSUE THREAD

**Issue by Emporea** [mileskrell/text-torch#2]
This app wont import any messages when using sms&mms with signal.

Is there any possibility to get this working? This would be awesome, also if it included the signal's own messages.

Happy New Year btw. :)

**Comment by uber97** [comment]
+1

I'm an avid Signal user. Would be awesome if it'll support Signal messages (not SMS)!

**Comment by mileskrell** [comment]
This would certainly be a nice feature to have. I'll take a look at it when I have time.

**Comment by mileskrell** [comment]
Agreed, Signal support would certainly be neat. I'll look into it as time permits 🙂

Thanks for opening this issue!

**Comment by fl0pedro** [comment]
I am not sure how well Text Torch would be able to interface with Signal messages, but maybe you can do something that analyzes the encrypted backups. Here is a project that decrypts it if you want to check that out: https://github.com/xeals/signal-back.

**Comment by lpstandardrp** [comment]
+1
 just came here wondering about Signal message analysis...

> I am not sure how well Text Torch would be able to interface with Signal messages, but maybe you can do something that analyzes the encrypted backups. Here is a project that decrypts it if you want to check that out: https://github.com/xeals/signal-back.

How might you or someone go about this once you have the decrypted Signal Backup file(s)? Export to XML and then import that XML into SMS Backup & Restore to "populate" phone's SMS with Signal's messages and then analyze using Text Torch? 
For reference: (Taken from https://github.com/xeals/signal-back):
> `signal-back` is intended to use the same decryption process as the Signal app uses when importing its backups, to make them readable without being used by the app.
> Usage... (build or download pre-built...)...
> `Usage: signal-back COMMAND [OPTION...] BACKUPFILE` ... ('export')...
> Current export formats are:
> 
> XML: 
> Targeted to be compatible with [SMS Backup & Restore](https://play.google.com/store/apps/details?id=com.riteshsahu.SMSBackupRestore) 
> CSV
> Go structure representation ("raw")

1. Export as XML
2. Import XML into SMS Backup and Restore
3. Use Text Torch to Analyze

Seem right?

Can this be accompliahed on Android alone? (i'm pretty sure a Samsung Galaxy Note 9 is or could fit the "Linux, and 32-bit (386) or 64-bit (amd64)" criteria (*see quote below*)... is that correct? If so, how would you verify which binary will work and how would you recommend exrcuting(?)(Correct verbiage- 'executing'?) The prebuilt-binary on aforementiined Samsung Android phone? 
Termux? 
NixOS?
 UserLand? 

*quote* from: https://github.com/xeals/signal-back#example-usage
> Example usage

> Download whichever binary suits your system from the [releases page](https://github.com/xeals/signal-back/releases) ; Windows, Mac OS (darwin), or Linux, and 32-bit (386) or 64-bit (amd64).
